### ISSUE TITLE
Add comparison to the API

### ISSUE THREAD

**Issue by vsoch** [NeuroVault/NeuroVault#207]
If we eventually want comparison to drive meta analysis, a user should be able to find the most similar images programmatically (via the API)

**Comment by erramuzpe** [comment]
@chrisfilo I am trying to solve this and I made some progress. I can retrieve the info for a specific comparison information using the API, but manually. I mean, I statically select an image, and gave the info about comparisons through the API. 

```
class ComparisonViewSet(viewsets.ModelViewSet):

    max_results = 100
    queryset = get_existing_comparisons(PK=3).extra(select={
          "abs_score": "abs(similarity_score)"}).
          order_by("-abs_score")[0:max_results]  # NOTE PK=3 -> this is the "static"
    serializer_class = ComparisonSerializer
    permission_classes = (ObjectOnlyPolymorphicPermissions,)
```
```
class ComparisonSerializer(serializers.Serializer):
    id = serializers.ReadOnlyField(source='image2.pk')
    abs_score = serializers.ReadOnlyField()
```

with this, I get the paired information in `http://127.0.0.1/api/comparison` , great. BUT I encountered some troubles when trying to extend this for `http://127.0.0.1/api/comparison/{pk}` . There are 2 kind of problems.

- The first one has relation with the way of retrieving the information and how the REST API works. As I have understood, the API works with a general query; say `/api/comparison` and then, you can filter it with using `/api/comparison/{pk}` using the `@detail_route` method or a retrieve function, right? This is prepared for a Model/QuerySet based architecture. The problem here is that we don't need the first part, just the part where the` {pk}` is used. I can not make it work in this way as is asking me a first query.

- The second one is conceptual; so, even if I manage to solve the first problem for the actual implementation, the aim is to substitute the Comparison model for an Engine that can be queried, but we are not going to get responses as QueryObjects/Models and the integration in the API looks difficult to adapt. My question is: is there any way of building a ViewSet for the API without using the Model architecture and getting rid of  `/api/comparison` ? 

Sorry for the mess, I hope all this makes sense to you!

**Comment by chrisfilo** [comment]
You are right about the lack of need for the Comparison model in one of the implementations. I would recommend to create JSON replies without using the Django REST Framework (see how atlas and gene expression APIs are implemented). It would make replacing this implementation easier later on.

**Comment by erramuzpe** [comment]
Thanks!
